src/makeCalculator.js

```diff
@@ -4,7 +4,35 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+    add(num) {
+      this.result += num;
+    },
+    subtract(num) {
+      this.result -= num;
+    },
+    multiply(num) {
+      this.result *= num;
+    },
+    divide(num) {
+      if (num !== 0) {
+        this.result /= num;
+      } else {
+        return 'Cannot divide by zero!';
+      }
+    },
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+    operate(callback, num) {
+      callback.call(this, num);
+
+      return this;
```

Comment:
[CODE KNOWLEDGE] - The `operate` method should include a check to ensure that the passed `callback` argument is indeed a function using `typeof callback === 'function'`. This is to prevent runtime errors if a non-function is passed as a callback.